File: Haskell/utils/PuzzleUtils.hs
module PuzzleUtils
  (puzzle_report) where


puzzle_report :: (Show input, Show output) => Int -> Int -> String -> input -> output -> IO ()
puzzle_report puzzle_number puzzle_part puzzle_name input output = do
    putStrLn ""
    putStrLn "=========================================================="
    putStrLn $ "Puzzle Number: " ++ show puzzle_number ++ " | Part: " ++ show puzzle_part
    putStrLn $ "  Puzzle Name: " ++ show puzzle_name
    putStrLn "----------------------------------------------------------"
    putStrLn "Input: "
    putStrLn $ show input
    putStrLn "----------------------------------------------------------"
    putStrLn "Result: "
    putStrLn $ show output
    putStrLn "=========================================================="
    putStrLn ""

File: Rust/src/utils/puzzle_utils.rs

//use std::fmt::Display;
use std::fmt::Debug;

pub fn report<I : Debug,O : Debug>(puzzle_number : i32, puzzle_part : i32, puzzle_name : &str, input : &I, output : &O)
{
    println!("");
    println!("==========================================================");
    println!("Puzzle Number: {puzzle_number} | Part: {puzzle_part}");
    println!("  Puzzle Name: {puzzle_name}");
    println!("----------------------------------------------------------");
    println!("Input: ");
    println!("{input:?}");
    println!("----------------------------------------------------------");
    println!("Output: ");
    println!("{output:?}");
    println!("==========================================================");
}


